pkgs/servers/consul/default.nix

```diff
@@ -1,30 +1,20 @@
-{ stdenv, buildGoPackage, fetchFromGitHub }:
+{ stdenv, buildGoModule, fetchFromGitHub }:
 
-buildGoPackage rec {
+buildGoModule rec {
   pname = "consul";
-  version = "1.6.2";
+  version = "1.7.2";
   rev = "v${version}";
 
-  goPackagePath = "github.com/hashicorp/consul";
-
-  # Note: Currently only release tags are supported, because they have the Consul UI
-  # vendored. See
-  #   https://github.com/NixOS/nixpkgs/pull/48714#issuecomment-433454834
-  # If you want to use a non-release commit as `src`, you probably want to improve
-  # this derivation so that it can build the UI's JavaScript from source.
-  # See https://github.com/NixOS/nixpkgs/pull/49082 for something like that.
-  # Or, if you want to patch something that doesn't touch the UI, you may want
-  # to apply your changes as patches on top of a release commit.
   src = fetchFromGitHub {
     owner = "hashicorp";
     repo = pname;
     inherit rev;
-    sha256 = "0r9wqxhgspgypvp9xdv931r8g28gjg9njdignp84rrbxljix25my";
+    sha256 = "1q587d8aqfkwg4fymr56fnf038vkxbdqz5yilz96dzny27dhspj4";
   };
 
-  preBuild = ''
-    buildFlagsArray+=("-ldflags" "-X github.com/hashicorp/consul/version.GitDescribe=v${version} -X github.com/hashicorp/consul/version.Version=${version} -X github.com/hashicorp/consul/version.VersionPrerelease=")
-  '';
```

Comment:
Could you comment (here, or even better in the commit message) whether/why consul no longer needs this passed to know its own git version?

Reply:
It builds without it - I just deleted all customizations and produced the minimum building version.

Reply:
Added lines back in because I don't have strong opinions.

Reply:
> It builds without it - I just deleted all customizations and produced the minimum building version.

I don't think that is a good approach in general -- if you check out the `git blame` it shows that this fixed real issues (#24606; users expect that a program's `--version` works).

I do think that commit should have had a comment though -- my bad!

> Added lines back in because I don't have strong opinions.

:+1: